core/trino-main/src/main/java/io/trino/execution/scheduler/BinPackingNodeAllocatorService.java

```diff
@@ -575,7 +575,7 @@ public ReserveResult tryReserve(PendingAcquire acquire)
                     .max(comparator)
                     .orElseThrow();
 
-            if (nodesRemainingMemoryRuntimeAdjusted.get(selectedNode.getNodeIdentifier()) >= acquire.getMemoryLease() || isNodeEmpty(selectedNode.getNodeIdentifier())) {
+            if (acquire.getMemoryLease() == 0 || nodesRemainingMemoryRuntimeAdjusted.get(selectedNode.getNodeIdentifier()) >= acquire.getMemoryLease() || isNodeEmpty(selectedNode.getNodeIdentifier())) {
```

Comment:
if  `getMemoryLease ==0`, then `nodesRemainingMemoryRuntimeAdjusted.. >= getMemoryLease`, right?

is the change necessary? if so, this means nodesRemainingMemoryRuntimeAdjusted can be negative. document

Reply:
Yeah - it can be negative right now